src/items/DocumentItem.tsx

```diff
@@ -24,9 +23,11 @@ export interface DocumentItemProps {
   saveButtonText?: string;
   showActions?: boolean;
   showEmpty?: boolean;
-  showCollapse?: boolean;
   styles?: React.CSSProperties;
-}
+} & (
+  | { showCollapse: true; item: DocumentItemTypeRecord }
+  | { showCollapse?: false; item: Pick<DocumentItemType, 'extra'> }
```

Comment:
I am curious whether it works if you don't have exactly the same type, since you are marking `showCollapse` as optional.

Reply:
Yes, for the compiler this is the same as `showCollapse: true | false | undefined`. So `undefined` is handled by the second type.